eng/test.yaml

```diff
@@ -0,0 +1,58 @@
+steps: 
+
+- task: DotNetCoreCLI@2
+  displayName: Restore (dotnet restore)
```

Comment:
Why do we need to restore again? We already did the restore during the build

Reply:
Without the restore, the `dotnet test` step (next) cannot find the Microsoft.NET.Test.Sdk.

Reply:
Is this really the way to do this though? the sdk has already been restored during the build, Does the dotnet test step take a path that we can point to where we installed in the first place?

Reply:
When I attempted to run `dotnet test` without the restore stage (and passing in the --no-build parameter), it complained that it couldn't find the specific version of Microsoft.NET.Test.Sdk that is needed to collect code coverage (16.3.0). I realize that the restore is done immediately before in the build step, so it doesn't make sense to me why these files aren't available in subsequent steps, too. 

@alexperovich, do you have any insight as to why this is the case? I don't recall needing to do this for Helix, so maybe it's another .NET Core-ism?

Reply:
@riarenas do you know if there's anything in the cibuild.cmd that does any sort of clean up? Looks like the last thing that happens on the build/publish stage is signing, but I don't know if there's some undocumented clean up going on as to why the restored dependencies would be missing for subsequent steps.

Reply:
@mjanecke its not a .NET core ism, its an Arcadeism. It changes things all over with how stuff is built/restored, and doesn't configure the environment properly for future steps to work. Perhaps the arcade build is setting NUGET_PACKAGES to some other folder and its not getting preserved for future steps so they look in the normal place.

Reply:
Yep, that is the problem. This line https://github.com/dotnet/arcade-services/blob/master/eng/common/tools.ps1#L458 is setting the NUGET_PACKAGES environment variable to a local folder, and the restore is restoring packages to that location, but it doesn't set NUGET_PACKAGES as a build variable so the dotnet test command doesn't have that custom location set and fails because there is nothing in the default location.

Reply:
Okay, I'll play around with that on Monday to see if I can retain the restored packages from the build step.

Reply:
Woot woot, got this to pull the packages from the correct location.